### DISCUSSION TITLE
DDEV addons - archived and forked repositories

### DISCUSSION THREAD

**Discussion by ChloeSartorelli** [ddev/ddev#6643] [Ideas]
Right now, an addon in a repository which has been archived doesn't indicate that it's abandoned.

There are two options I can think of for this:
1. don't display it in the `ddev add-on list --all` list at all, but still allow it to be installed. Then, on installation, output some "this add-on is in an archived repository" message.
2. add some way to indicate the add-on is abandoned when it's displayed in `ddev add-on list --all`. As above, add a message on installation.

The other scenario I thought of after I noticed this is forked repositories. I haven't checked this, but _presumably_ forks display in the addon list? If they do... should they? If they don't, then I assume this has already been discussed and thought about.

**Comment by tyler36** [comment]
I think retaining the ability to install/remove archive add-on is important.
Perhaps we simple don't show them in _any_ listing. Out of sight, out of mind, but still available if you know where to look.

> The other scenario I thought of after I noticed this is forked repositories. 

Forks do NOT appear in the lists. If they did, you would see a ton of addons from @GuySartorelli as they raining down document update PRs at the moment. (Thank you!)

**Reply by ChloeSartorelli** [reply]
> I think retaining the ability to install/remove archive add-on is important.

Completely agree. I think some sort of message should be output, but definitely still allow them to be installed. This is more about whether and how they appear in the list.

**Reply by rfay** [reply]
First, @GuySartorelli THANK YOU FOR THE INCREDIBLE WORK, first for the ambitious change and now for the ambitious cleanup!!! Craziness, and so much appreciated.

Second, if a repo does not have the `ddev-get` topic it doesn't get discovered by `ddev add-on list --all`, so the first line of defense is to get the author to remove the label. The second line of defense, for archived, as @stasadev says below, we can exclude `archived`.

**Reply by rfay** [reply]
@tyler36 :
> I think retaining the ability to install/remove archive add-on is important.

The ability to install or remove is not related to the visibility in the github api. Removal is done by the stored information in the .ddev config, and installation can be done via github short-link or full link at any time, even if archived.

**Reply by tyler36** [reply]
> Second, if a repo does not have the ddev-get topic it doesn't get discovered by ddev add-on list --all, so the first line of defense is to get the author to remove the label. The second line of defense, for archived, as @stasadev says below, we can exclude archived.

We can't always count on maintaining being willing or able to remove the label; especially on "abandoned" projects. Ideally, any script should include an "exclude" list that are never display.

**Comment by stasadev** [comment]
@GuySartorelli Thank you very much for all these updates!

> Perhaps we simple don't show them in *any* listing.

We can exclude archived from the list with `archived:false`, I see there is one archive now:

https://api.github.com/search/repositories?q=topic:ddev-get+fork:true+archived:true

**Reply by ChloeSartorelli** [reply]
Sweet, that sounds sensible. There's no rush to do this, so maybe it can be part of the work to [introduce a registry](https://github.com/ddev/ddev/issues/6383)?

**Reply by rfay** [reply]
It's actually just in the query, PR is welcome.  I could be wrong but I think it's just a minor tweak to ListAvailableAddons(), https://github.com/ddev/ddev/blob/fae659e4d94cb2e70568c420046360e184a50982/pkg/ddevapp/addons.go#L160-L195
